src/types/entity_id.ts

```diff
@@ -0,0 +1,25 @@
+const entityIdRegex = /^([a-f]|[0-9]){8}-([a-f]|[0-9]){4}-([a-f]|[0-9]){4}-([a-f]|[0-9]){4}-([a-f]|[0-9]){12}$/;
+
+export class EntityID {
+	constructor(private readonly entityId: string) {
+		if (!entityId.match(entityIdRegex)) {
+			throw new Error(`Invalid entity ID '${entityId}'!'`);
+		}
+	}
+
+	toString(): string {
+		return this.entityId;
+	}
+
+	valueOf(): string {
+		return this.entityId;
+	}
+
+	isEqualTo(entityId: EntityID): boolean {
+		return `${this.entityId}` === `${entityId}`;
```

Comment:
`this.entityId` is already a primitive string
```suggestion
		return this.entityId === `${entityId}`;
```

Reply:
This is also valid
```suggestion
		return `${this}` === `${entityId}`;
```

Reply:
This is redundant, but it's not the end of the world